### PR TITLE
Better fix for `new-for-builtins`

### PR DESCRIPTION
**Pull request by fisker**
1. Fix some edge case.
2. Keep comments.

### CODE REVIEW

File: test/snapshots/new-for-builtins.js.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        |                ^^^^^^^^^^ Use `new Object()` instead of `Object()`.␊
+    `
+
+## Invalid #2
+      1 | const symbol = new (Symbol)("");
+
+> Output
+
+    `␊
+      1 | const symbol = (Symbol)("");␊
```
Comment (fisker) [inline test/snapshots/new-for-builtins.js.md:29]: This was fixed to

```js
const symbol = Symbol)("");␊
```

File: test/snapshots/new-for-builtins.js.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        |                ^^^^^^^^^^^^^^^^ Use `Symbol()` instead of `new Symbol()`.␊
+    `
+
+## Invalid #3
+      1 | const symbol = new /* comment */ Symbol("");
+
+> Output
+
+    `␊
+      1 | const symbol = /* comment */ Symbol("");␊
```
Comment (fisker) [inline test/snapshots/new-for-builtins.js.md:45]: This comment was removed.

File: test/snapshots/new-for-builtins.js.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `Symbol()` instead of `new Symbol()`.␊
+    `
+
+## Invalid #4
+      1 | const symbol = new Symbol;
+
+> Output
+
+    `␊
+      1 | const symbol = Symbol();␊
```
Comment (fisker) [inline test/snapshots/new-for-builtins.js.md:61]: This was fixed to

```js
const symbol = Symbol;
```
